packages/beelzebub/data_stream/logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -170,53 +170,70 @@ processors:
       extract_device_type: true
       ignore_missing: true
 
-  - gsub:
-      if: ctx.json?.event?.Headers != null && ctx.json?.event?.Headers != ""
-      description: "Replace leading and trailing garbage to support KVP extraction"
+### NOTE: Beelzebub now produces logs with two fields for headers, one is a string and the other is an object. The following 7 processors are used to support both 
+#         log formats/situations and to avoid creating a type conflict with existing indices while still ensuring we have usable headers and header values to search with.
+
+  - remove:
+      if: ctx.json?.event?.Headers != null && ctx.json.event.Headers instanceof String && ctx.json.event.Headers == ""
```

```suggestion
      if: ctx.json?.event?.Headers == ""
```
In order for `ctx.json?.event?.Headers == ""`, `ctx.json?.event?.Headers instanceof String` must be true, and for this to be true, `ctx.json?.event?.Headers != null`, so we can just check the literal string identity.

However, this is not what the descriptions says; it says
```suggestion
      if: ctx.json?.event?.Headers == null || ctx.json.event.Headers == ""
```

On that basis, I suppose you want the second suggestion.